jamf-upload.sh

```diff
@@ -199,10 +199,12 @@ processors_directory="$DIR/JamfUploaderProcessors"
 ## ARGUMENTS ##
 ###############
 
-echo "" > "$temp_processor_plist"  # ensure an empty processor at the start of the run
+rm -rf "$temp_processor_plist" # delete any existing file (or folder) at temp_processor_plist path
+/usr/libexec/PlistBuddy -c 'Clear dict' "$temp_processor_plist" # ensure an empty processor at the start of the run
```

Comment:
Lines 202 seems unnecessary given line 203, or am I missing something?

Reply:
It's a bit over cautious, but it's a habit I've gotten into since if a _folder_ happens to exist in the location you want to write a temporary _file_ to, all file creating/editing techniques would fail until the folder has been deleted.

Reply:
I also just found that if the file is an empty 0 byte file that `PlistBuddy -c 'Clear dict'` will also fail. So deleting any existing file (or folder) is the safest option to guarantee a valid plist gets created.

```
% mkdir ~/Desktop/test.plist                                  
% /usr/libexec/PlistBuddy -c 'Clear dict' ~/Desktop/test.plist
Error Reading File: /Users/pico/Desktop/test.plist

% rm -rf ~/Desktop/test.plist

% touch ~/Desktop/test.plist                                  
% /usr/libexec/PlistBuddy -c 'Clear dict' ~/Desktop/test.plist
Cannot parse a NULL or zero-length data
Error Reading File: /Users/pico/Desktop/test.plist

% rm -rf ~/Desktop/test.plist

% /usr/libexec/PlistBuddy -c 'Clear dict' ~/Desktop/test.plist
File Doesn't Exist, Will Create: /Users/pico/Desktop/test.plist
Initializing Plist...
```